EasyCommands/CommandParsers/ParameterProcessors.cs

```diff
@@ -457,16 +458,24 @@ partial class Program : MyGridProgram {
                     processorIndex = 0;
                     continue;
                 }
-                if (!revisit) sortedParameterProcessors.RemoveAt(processorIndex);
-                else processorIndex++;
+                if (!revisit) {
+                    sortedParameterProcessors.Remove(current);
+                    processorRanks.Remove(current.Rank);
+                } else processorIndex++;
             }
 
             return branches;
         }
 
-        void AddProcessors(List<CommandParameter> types, List<ParameterProcessor> sortedParameterProcessors) {
-            sortedParameterProcessors.AddRange(types.SelectMany(t => parameterProcessorsByParameterType.ContainsKey(t.GetType()) ? parameterProcessorsByParameterType[t.GetType()] : NewList<ParameterProcessor>()).ToList());
-            sortedParameterProcessors.Sort();
+        void AddProcessors(List<CommandParameter> types, SortedList<ParameterProcessor, int> sortedParameterProcessors, HashSet<int> processorRanks) {
+            var processors = types.Select(t => t.GetType())
+                .SelectMany(t => parameterProcessorsByParameterType.GetValueOrDefault(t, NewList<ParameterProcessor>()))
+                .Where(p => !processorRanks.Contains(p.Rank));
+
+            foreach (ParameterProcessor processor in processors) {
+                sortedParameterProcessors[processor] = processor.Rank;
+                if (processorRanks.Contains(processor.Rank)) processorRanks.Add(processor.Rank);
```

Comment:
The contains check is unnecessary.  Turns out HashSet does the de-dupe in a nice way.